--- a/src/fluddokt/opsu/fake/Graphics.java
+++ b/src/fluddokt/opsu/fake/Graphics.java
@@ -60,8 +60,16 @@
 	public void setColor(Color ncolor) {
 		fgcolor = ncolor;
-		batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
-		shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
+		if (mode == SPRITE) {
+			batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
+		} else if (mode == SHAPEFILLED || mode == SHAPELINE) {
+			shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
+		} else {
+			batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
+			shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a);
+		}
 	}
 	
 	public void setColorAlpha(Color ncolor, float alpha) {
 		fgcolor = ncolor;
-		batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
-		shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
+		if (mode == SPRITE) {
+			batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
+		} else if (mode == SHAPEFILLED || mode == SHAPELINE) {
+			shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
+		} else {
+			batch.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
+			shapeRender.setColor(fgcolor.r, fgcolor.g, fgcolor.b, fgcolor.a * alpha);
+		}
 	}
